webrtc.html

```diff
@@ -625,6 +625,313 @@
+        <code><a>RTCPeerConnection</a></code> instance <var>connection</var> to
+        <var>newState</var>, the User Agent MUST queue a task that sets
+        <var>connection</var>'s <a href=
+        "#dom-peerconnection-ice-gathering-state">iceGatheringState</a>
+        attribute to <var>newState</var> and fires a simple event named
+        <code><a href=
+        "#event-icegatheringstatechange">icegatheringstatechange</a></code> at
+        <var>connection</var>.</p>
+
+        <p>To <dfn id=
+        "update-ice-connection-state">update the ice connection state</dfn> of an
+        <code><a>RTCPeerConnection</a></code> instance <var>connection</var> to
+        <var>newState</var>, the User Agent MUST queue a task that sets
+        <var>connection</var>'s <a href=
+        "#dom-peerconnection-ice-connection-state">iceConnectionState</a>
+        attribute to <var>newState</var> and fires a simple event named
+        <code><a href=
+        "#event-iceconnectionstatechange">icegatheringstatechange</a></code> at
+        <var>connection</var>.</p>
+
+        <p>To <dfn id="set-description">set an RTCSessionDescription</dfn>
+        <var>description</var> on an <code><a>RTCPeerConnection</a></code>
+        object <var>connection</var>, run the following steps:</p>
+
+        <ul>
+          <li>
+            <p>When the setter method is invoked, the user agent MUST run the
+            following steps:</p>
+
+            <ol>
+              <li>
+                <p>If <var>connection</var>'s
+                <a href="#dom-peerconnection-signaling-state">signaling
+                state</a> is <code>closed</code>, the user agent MUST return a
+                promise rejected with an <code>InvalidStateError</code>.</p>
+              </li>
+
+              <li>
+                <p>If <var>description</var> is set as a local description and
+                contains a different set of ICE credentials, then the ICE Agent
+                MUST trigger an ICE restart, and run the following steps:</p>
+
+                <ol>
+                  <li>
+                    <p>If <var>connection</var> is not already gathering,
+                    <a href="#update-ice-gathering-state">update
+                    <var>connection</var>'s ice gathering state</a> to
+                    <code>gathering</code>.</p>
+                  </li>
+
+                  <li>
+                    <p>If <var>connection</var>'s <a href=
+                    "#dom-peerconnection-ice-connection-state">ice connection state</a>
+                    is <code>completed</code>, <a href=
+                    "#update-ice-connection-state">update
+                    <var>connection</var>'s ice connection state</a> to
+                    <code>connected</code>.</p>
+                  </li>
+                </ol>
+              </li>
+
+              <li>
+                <p>The user agent MUST start the process to apply
+                <var>description</var>.</p>
```

Comment:
Note that when you run in parallel, you have two things to do:
1. await the completion of any other tasks that have been enqueued.
2. check that the pc is not closed
3. then apply the description